doc/source/developer_overview.rst

```diff
@@ -210,115 +213,123 @@ For a complete definition of the ``rcl`` |API|, see the |API| documentation:
+The ``rosidl`` repository, located on |GitHub|_ at `ros2/rosidl <https://github.com/ros2/rosidl>`_, defines the message IDL syntax, i.e. syntax of ``.msg`` files, ``.srv`` files, etc., and contains |packages| for parsing the files, for providing CMake infrastructure to generate code from the messages, for generating implementation agnostic code (headers and source files), and for establishing the default set of generators.
 The repository contains these |packages|:
 
--  ``rosidl_cmake``: provides CMake functions and Modules for generating code from ``rosidl`` files, e.g. ``.msg``, ``.srv``, etc.
--  ``rosidl_default_generators``: defines the list of default generators which ensures that they are installed as dependencies, but other injected generators will also be used.
+-  ``rosidl_cmake``: provides CMake functions and Modules for generating code from ``rosidl`` files, e.g. ``.msg`` files, ``.srv`` files, etc.
+-  ``rosidl_default_generators``: defines the list of default generators which ensures that they are installed as dependencies, but other injected generators can also be used.
 -  ``rosidl_generator_c``: provides tools to generate C header files (``.h``) for ``rosidl`` files.
 -  ``rosidl_generator_cpp``: provides tools to generate C++ header files (``.hpp``) for ``rosidl`` files.
+-  ``rosidl_generator_py``: provides tools to generate Python modules for ``rosidl`` files.
 -  ``rosidl_parser``: provides Python |API| for parsing ``rosidl`` files.
 
-Generators for other languages, e.g. ``rosidl_generator_py``, are hosted externally (in a different repository) but would use the same mechanism that ``rosidl_generator_c`` and ``rosidl_generator_cpp`` use to "register" themselves as a ``rosidl`` generator.
+Generators for other languages, e.g. ``rosidl_generator_java``, are hosted externally (in different repositories) but would use the same mechanism that the above generators use to "register" themselves as a ``rosidl`` generator.
+
+In addition to the aforementioned |packages| for parsing and generating headers for the ``rosidl`` files, the ``rosidl`` repository also contains |packages| concerned with "type support" for the message types defined in the files.
+Type support refers to the ability to interpret and manipulate the information represented by ROS message instances of particular types (serialization and deserialization, for example).
+Type support can either be provided by code that is generated at compile time or it can be done programmatically based on the contents of the ``rosidl`` file and the data received, by introspecting the data.
+In the case of the latter, where type support is done through runtime interpretation of the messages, the message code generated by ROS 2 can be agnostic to the rmw implementation.
```

Comment:
In both cases code is generated. Currently, there is no runtime interpretation of the msg files. In the "static" case, code to do the work (register type, publish, take, serialize, deserialize) is generated for each message type. In the "dynamic" case, meta information is generated (at compile time) from the .msg files and that is used with generic functions to do the work.

What you've described is one step further than the second case, where meta information is used with the generic functions like before, but also the meta information is created at runtime by parsing the msg file.

I'll update this section, but this comment is more to let you know what I'm planning on doing.